03_launch_mgmt_cluster.sh

```diff
@@ -551,12 +551,75 @@ cat <<EOF > tilt-settings.json
   }
 }
 EOF
+sed -i 's/yaml = str(kustomizesub(context + "\/config"))/yaml = str(kustomizesub(context + "\/config\/tls"))/' Tiltfile
 make kind-reset
 kind create cluster --name capm3 --image="kindest/node:${KUBERNETES_VERSION}"
 kubectl create namespace "${NAMESPACE}"
+kubectl create namespace baremetal-operator-system
 mkdir -p "${HOME}/.cluster-api/overrides/infrastructure-metal3/${CAPM3RELEASE}"
-sleep 120 && launch_ironic &
-sleep 120 && apply_bm_hosts &
-make tilt-up
+make tilt-up &
+# wait for cert-manager to be ready
+sleep 120
+launch_ironic
+# deploy bmo in order to generate ironic credentials and tls
+launch_baremetal_operator
+apply_bm_hosts
+# remove bmo, so that is deployed and monitored by Tilt
+kubectl delete deployments.apps -n baremetal-operator-system baremetal-operator-controller-manager
+
+pushd "${BMOPATH}"
+# shellcheck disable=SC2155
+# shellcheck disable=SC1001
+export IRONIC_SECRET_NAME=$(kubectl get secrets  -n baremetal-operator-system -oname | grep ironic-credentials | cut -f2 -d\/)
+# shellcheck disable=SC2155
+# shellcheck disable=SC1001
+export IRONICINSPECTOR_SECRET_NAME=$(kubectl get secrets  -n baremetal-operator-system -oname | grep "ironic-inspector-credentials" | cut -f2 -d\/)
+
+# use existing ironic and inspector secrets 
+# Tilt cannot generate new credentials and certificates as it would mismatch with what ironic is already configured with
+cat <<EOF >> config/tls/tls_ca_patch.yaml
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        volumeMounts:
+          - mountPath: /opt/metal3/auth/ironic
+            name: ironic-credentials
+            readOnly: true
+          - mountPath: /opt/metal3/auth/ironic-inspector
+            name: ironic-inspector-credentials
+            readOnly: true
+      volumes:  
+      - name: ironic-credentials
+        secret:
+          secretName: ${IRONICINSPECTOR_SECRET_NAME}
+      - name: ironic-inspector-credentials
+        secret: 
+          secretName: ${IRONIC_SECRET_NAME}
+EOF
+tools/bin/kustomize build config/tls/
+popd
+
+# Start watching changes on bmo
+cat <<EOF > tilt-settings.json
+{
+  "provider_repos": [ "../baremetal-operator", "../ip-address-manager"],
+  "enable_providers": [ "metal3-bmo", "metal3-ipam"],
```

Comment:
not a blocker, but have a question on these. If I am not wrong, the intention is to have all components present in Tilt and I think we are missing CAPI components (https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/docs/dev-setup.md#tilt-for-dev-in-both-capm3-and-capi) since we are adding IPAM/BMO (https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/docs/dev-setup.md#including-baremetal-operator-and-ip-address-manager)?

Reply:
There are components that are deployed. With the current version, the capi and cert manager are deployed. 
```
cluster-api-provider-metal3/Tiltfile 
	deploy_cert_manager()
	deploy_capi()
```

Reply:
Yes, I was talking about adding CAPI controllers (docker", "kubeadm-bootstrap", "kubeadm-control-plane) to Tilt itself as we are adding CAPM3/IPAM/BMO controllers

Reply:
True, they are not visible on the dashboard. That requires a change on Tilt or related files on the capm3 side.
Looking into that now